components/script/dom/performance.rs

```diff
@@ -41,6 +103,79 @@ impl Performance {
                            window,
                            PerformanceBinding::Wrap)
     }
+
+    /// Add a PerformanceObserver to the list of observers with a set of
+    /// observed entry types.
+    pub fn add_observer(&self,
+                        observer: &DOMPerformanceObserver,
+                        entry_types: Vec<DOMString>) {
+        self.observers.borrow_mut().push(PerformanceObserver {
```

Comment:
According to https://w3c.github.io/performance-timeline/#dom-performanceobserver-observe() we should be replacing any existing matching one.